src/databases/Silo/avtSiloFileFormat.C

```diff
@@ -14491,19 +14491,50 @@ avtSiloFileFormat::GetExternalFacelist(int dom, const char *mesh)
 //    Moved debug5 output lines indicating that global node ids are being read
 //    down so that it appears in debug logs only when there is actually data
 //    being read and returned from the plugin.
-// ****************************************************************************
+//
+//    Mark C. Miller, Mon Feb 12 20:26:21 PST 2024
+//    Refactored away GetGlobalZoneIds and GetGlobalNodeIds into this one
+//    method that does both and does so for both UCD and POINT meshes.
+//    The main tricks are to a) set Silo's data read mask appropriately (and
+//    then also reset it when done) and then b) to allow VisIt's cache to take
+//    ownership of the resulting data. We do this by zeroing the Silo object's
+//    data member holding the global ids and keeping a separate local pointer
+//    to the data. For point meshes, nodes and zones are one in the same so the
+//    logic for both is the same. For UCD meshes, there is separate logic for
+//    global node ids and global zone ids embodied by a different set of args
+//    to the LOAD_GLOBAL_IDS macro'd logic.
+// ****************************************************************************
+#define LOAD_GLOBAL_IDS(DBTYP, GETFUNC, CKMEMBR, FREEFUNC, MASK, GIDTYP, NIDS, IDS) \
```

Comment:
can we use a templated function instead of a macro?

Reply:
I don't think so...we de-refing differently named data members on the Silo object struct pointer.

Reply:
got it, macro is a bit scary -- it's a short amount code, are there just two cases - If so, could we just implement them?

Reply:
Its actually 3 cases. And, it means imlementing GetGloblaIds three times where 90+% of the code is the same. Thats a maint. issue. I am not to scared of macros. We use them all over the place. For situations like this, they make a lot of sense. I did minimize the amount of code that is macro'd